packages/pigeon/lib/dart_generator.dart

```diff
@@ -140,11 +140,13 @@ void _writeFlutterApi(
                 indent.writeln('// ignore message');
                 call = 'api.${func.name}()';
               } else {
+                if (!opt.isNullSafe) {
```

Comment:
I don't believe this is correct.  If we are generating code with null safety, message will be an `Object?`, so the null check would still be needed here.  I don't think we have a test that covers this unfortunately.  I think the only test that runs this is non-null-safe.  You can see that here: https://github.com/flutter/packages/blob/344f575216cacfcf09d477cf2a32fbe1fac574d3/packages/pigeon/mock_handler_tester/test/widget_test.dart#L70

Reply:
Yeah it will be `Object?` in null safe mode. But then we don't need to check for `null` since `null` is valid value. It will be passed to `$argType.decode(Object? message)` and there the `null` message will be handled correctly. Or do I misunderstand something here?

Reply:
There might be merit to what you are saying.  The problem is this PR has enough changes already.  I'm putting this back the way it was, including the test and signature for decode.  You can file this in a different PR after if you think it's the right change.